### PR TITLE
Admins should be able to send users a welcome email

### LINKED ISSUE #2189: Admins should be able to send users a welcome email
## Affected Area
User management

## Requested Feature
As a super/tenant admin I can see a list of users with actions. It would be good to have a 'Send invite' button which sent the user a welcome email. At the moment admins must inform users that they have accounts manually.

## Current Workaround
Admins must manually contact people after creating their accounts.

### PR DESCRIPTION
**Pull request by canstudios-nicolaw**
Fixes #2189

### CODE REVIEW

File: frontend/src/modules/userManagement/views/userView.js [hunk 1]
```diff
@@ -171,6 +172,23 @@ define(function(require){
...
 
+    onInviteClicked: function(e) {
+      Origin.Notify.confirm({
+        text: Origin.l10n.t('app.confirmsendinvite', { email: this.model.get('email') }),
+        callback: function(confirmed) {
+          if(!confirmed) {
+            return;
+          }
+
+          var $btn = $(e.currentTarget);
```
Comment (lc-thomasberger) [inline frontend/src/modules/userManagement/views/userView.js]: `e.currentTarget` is the userView element. `e.target` is the button element.

**Comment by lc-thomasberger** [comment]
Works fo rme

File: lib/usermanager.js [hunk 2]
```diff
@@ -832,6 +832,37 @@ exports = module.exports = {
...
+            return res.json(err);
+          }
+          var inviterDetails = userObject.email;
+          if (userObject.firstName && userObject.lastName) {
+            inviterDetails = userObject.firstName + ' ' + userObject.lastName + ' (' + inviterDetails + ')';
+          }
+          var subject = app.polyglot.t('app.emailinvitesubject');
+          var body = app.polyglot.t('app.emailinvitebody', { inviter: inviterDetails, rootUrl: configuration.getConfig('rootUrl'), resetUrl: configuration.getConfig('rootUrl') + '/#user/forgot' });
+          app.mailer.send(req.body.email, subject, body, { name: 'emails/invite.hbs' }, function() {
+            res.statusCode = 200;
```
Comment (tomgreenfield) [inline lib/usermanager.js]: Please can we add in error handling here as per:
https://github.com/adaptlearning/adapt_authoring/blob/3f060704f7a94bf3d19f1f69f26c60cdfc5e2fa8/plugins/auth/local/index.js#L286-L288

File: routes/lang/en-application.json [hunk 3]
```diff
@@ -226,6 +228,8 @@
   "app.menupicker": "Menu picker",
   "app.menumanagement": "Manage Available Menus",
   "app.tenantmanagement": "Tenant Management",
+  "app.emailinvitesubject": "Welcome to Adapt",
+  "app.emailinvitebody": "You have been invited to join by %{inviter}<br><br>To login, head to your <a href='%{rootUrl}' target='_blank'>Adapt authoring tool</a><br><br>If you do not know your password you should request a reset at: <a href='%{resetUrl}' target='_blank'>%{resetUrl}</a>",
```
Comment (tomgreenfield) [inline routes/lang/en-application.json]: Typo: `login` → `log in`.

File: lib/usermanager.js (hunk 2 above)
Comment (canstudios-nicolaw) [reply lib/usermanager.js]: :+1: sorry I missed it

**Pull request merged**